jmbitcoin/jmbitcoin/secp256k1_transaction.py

```diff
@@ -634,15 +643,38 @@ def serialize_script(script):
     else:
         return result
 
-
 def mk_multisig_script(pubs, k):
     """ Given a list of pubkeys and an integer k,
     construct a multisig script for k of N, where N is
     the length of the list `pubs`; script is returned
     as hex string.
     """
-    return serialize_script([k] + pubs + [len(pubs)]) + 'ae'
+    return serialize_script([k] + pubs + [len(pubs)]) \
+        + 'ae' #OP_CHECKMULTISIG
 
+def mk_freeze_script(pub, locktime):
+    """
+    Given a pubkey and locktime, create a script which can only be spent
+    after the locktime as passed using OP_CHECKLOCKTIMEVERIFY
+    """
+    if not isinstance(locktime, int):
+        raise TypeError("locktime must be int")
+    if not isinstance(pub, str):
```

Comment:
I'd prefer to fix the type expected by these new functions. Also prefer all-binary-in (after #536 everything is binary in wallet and jmbitcoin), at least. Understood you do have to worry about hex a bit in this now-legacy version of jmbitcoin (and it's damn annoying!).
Also given the sensitivity of this script creation, I'd strongly recommend a parse of the incoming pubkey serialization to ensure that it represents a valid `secp256k1.PublicKey` object. See `secp256k1_main.is_valid_pubkey()`.